Tasks/Maven/maventask.ts

```diff
@@ -227,14 +227,43 @@ function applyPmdGoals(mvnRun: trm.ToolRunner):void {
 }
 
 // Returns true if any one code analysis tool was enabled.
-function isCodeAnalysisReportRequired():boolean {
+function isCodeAnalysisUploadRequired():boolean {
     return tl.getInput('pmdAnalysisEnabled', true) == 'true';
 }
 
+// Take a result object and attempt to upload the XML and HTML reports as artifacts. Throws if fail.
+function uploadBuildArtifactsFromResult(analysisResult:ar.AnalysisResult):void {
+    // make sure the staging folder exists and is empty
+    tl.rmRF(stagingDir);
```

Comment:
Followed an example from the CopyPublishArtifact task. This should probably be changed to the path indicated by "build.artifactStagingDirectory", however, which is perhaps slightly more appropriate.

Accepted.